macros/src/main/scala/gestalt/macros/Optional.scala

```diff
@@ -0,0 +1,41 @@
+import scala.gestalt._
+
+import scala.annotation.unchecked.{ uncheckedVariance => uncheckVar }
+
+final class Optional[+A >: Null](val value: A) extends AnyVal {
+  def get: A = value
+  def isEmpty = value == null
+
+  def getOrElse[B >: A](alt: => B)(implicit m: toolbox.WeakTypeTag[A] @uncheckVar): B = meta {
+    import toolbox._
+
+    val temp = toolbox.fresh("_temp")
+    val tempIdent = Ident(temp)
+
+    q"""
+      val $temp = $prefix
+      if ($tempIdent.isEmpty) $alt else $tempIdent.value
+    """
+  }
+
+  def map[B >: Null](f: A => B)(implicit m: toolbox.WeakTypeTag[A] @uncheckVar): Optional[B] = meta {
+    import toolbox._
+
+    val Function(param :: Nil, body) = f
+    val tempValDef = ValDef(toolbox.fresh("_temp"), prefix)
+    val tempIdent = Ident(tempValDef.symbol)
```

Comment:
Is the tpd => tpd transform essential to this macro, or we can be just fine with untpd => untpd?